mono/utils/mono-context.h

```diff
@@ -678,4 +678,10 @@ void mono_sigctx_to_monoctx (void *sigctx, MonoContext *mctx);
  */
 void mono_monoctx_to_sigctx (MonoContext *mctx, void *sigctx);
 
+// Currently we only use ucontext on platforms where sigaction is defined.
+// If we ever need ucontext without sigaction, change this macro to defined(UCONTEXT_REG_RAX)
+#if defined(MONO_SIGNAL_USE_SIGACTION)
```

Comment:
This change is wrong, it introduces an extra define for no good reason.

A better fix is the following:

- remove MONO_SIGNAL_USE_SIGACTION
- check for HAVE_UCONTEXT_H instead.